make/compiler/Makefile.gnu

```diff
@@ -104,6 +116,42 @@ export GNU_GPP_SUPPORTS_STRICT_OVERFLOW = $(shell test $(GNU_GPP_MAJOR_VERSION)
 endif
 
 #
+# Specifying the language standard for gcc/g++ to use
+#
+# If the gcc and g++ versions both support standard atomics:
+#   For gcc version >= 5, just use the default, which is at least C11.
+#   For g++ version >= 6, just use the default, which is at least C++14.
+#   Otherwise the defaults are C90/C++98, so specify C11 and C++11.
+# If one or both don't support standard atomics:
+#   Specify C99/C++98.
+#
+ifeq ($(GNU_GCC_SUPPORTS_STDATOMICS),$(GNU_GPP_SUPPORTS_STDATOMICS))
+  ifeq ($(GNU_GCC_SUPPORTS_STDATOMICS),1)
+    ifeq ($(shell test $(GNU_GCC_MAJOR_VERSION) -lt 5; echo "$$?"),0)
+      GCC_STD = -std=c11
+    else
+      GCC_STD =
+    endif
+    ifeq ($(shell test $(GNU_GPP_MAJOR_VERSION) -lt 6; echo "$$?"),0)
+      GPP_STD = -std=c++11
```

Comment:
Ah, yes, I'd noticed the seemingly reversed tests and why they worked, but confused myself about what this one was testing!